server/src/main/java/org/elasticsearch/index/mapper/DynamicTemplate.java

```diff
@@ -181,21 +181,24 @@ static DynamicTemplate parse(String name, Map<String, Object> conf) throws Mappe
                 throw new IllegalArgumentException("Illegal dynamic template parameter: [" + propName + "]");
             }
         }
-
-        if (match == null && pathMatch == null && matchMappingType == null) {
-            throw new MapperParsingException("template must have match, path_match or match_mapping_type set " + conf.toString());
```

Comment:
relaxing this makes sense to me. I wonder though about the behavior in a mixed cluster, where only new nodes can support the newer format, and older nodes would not accept it. I assume things will fail, but where and how?

Reply:
We have the validation here https://github.com/elastic/elasticsearch/blob/00b92c15d6e66f8e6fe33953a03edf9107c31c3a/server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java#L393-L398.

Reply:
It's the [smallest](https://github.com/elastic/elasticsearch/blob/321f46e1877d40d987294f28362619a322d92af3/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java#L673) version of non-client nodes.

Reply:
@javanna I pushed https://github.com/elastic/elasticsearch/pull/69948/commits/aef927e301f68b67ea3f44a3a1cbaee866377b17 to use the min node version of the current state instead.